### PR TITLE
Directly build WebRender LineDisplayItem

### PR DESCRIPTION
**Pull request by pyfisch**
Remove unused SimpleMatrixDetection.

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22035)

### CODE REVIEW

File: components/layout/display_list/builder.rs [hunk 1]
```diff
@@ -2272,10 +2277,15 @@ impl FragmentDisplayListBuilding for Fragment {
...
+        // TODO(gw): Use a better estimate for wavy line thickness.
+        let wavy_line_thickness = (0.33 * base.bounds.size.height).ceil();
+        state.add_display_item(DisplayItem::Line(CommonDisplayItem::new(
+            base,
+            webrender_api::LineDisplayItem {
+                orientation: webrender_api::LineOrientation::Horizontal,
+                wavy_line_thickness,
+                color: color.to_layout(),
+                style: LineStyle::Solid,
         })));
```
Comment (emilio) [inline components/layout/display_list/builder.rs]: nit: Indentation here feels a bit weird, probably the curly brace should be on its own line before this one now, and same above.

**Review by emilio — APPROVED** [review]
Looks great!

**Comment by pyfisch** [comment]
@bors-servo retry

* Cannot allocate memory

bors-servo <notifications@github.com> schrieb am So., 28. Okt. 2018, 22:56:

**Comment by CYBAI** [comment]
@bors-servo retry

- oom in `linux-rel-wpt`
- in `linux-rel-css`
```
{"status": "CRASH", "group": "default", "message": null, "stack": null, "subtest": null, "test": "/html/syntax/parsing/html5lib_tests15.html?run_type=uri", "line": 199926, "action": "test_result", "expected": "OK"}
```

**Comment by CYBAI** [comment]
@bors-servo retry

```
{"status": "FAIL", "group": "default", "message": "/css/CSS2/selectors/sibling-selector-002.xht 4e6ec2a4e7504e5c4bb739eb4fad793251d3180f\n/css/CSS2/reference/filler-text-below-green.xht 89fbe26567dcb89de46762b75a6737796e0220f3\nTesting 4e6ec2a4e7504e5c4bb739eb4fad793251d3180f == 89fbe26567dcb89de46762b75a6737796e0220f3", "stack": null, "subtest": null, "test": "/css/CSS2/selectors/sibling-selector-002.xht", "line": 13147, "action": "test_result", "expected": "PASS"}
```

**Pull request merged**